base_local_planner/src/obstacle_cost_function.cpp

```diff
@@ -74,7 +80,7 @@ std::vector<geometry_msgs::Point> ObstacleCostFunction::getScaledFootprint(const
   if (scale != 0.0) {
     const bool fwd = traj.xv_ > 0;
     const double forward_inflation = scale * max_forward_inflation_;
-    const double sideward_inflation = scale * max_sideward_inflation_;
+    const double sideward_inflation = scale * max_sideward_inflation_scale_ * max_sideward_inflation_;
```

Comment:
F: this is a bit,,, weird,,,,  or confusing,,,  or both

scale * scale    -->  rename `scale` as something like `speed_scaling_factor`
max * max       -->  `max_sideward_inflation_scale_` is not really a max, right?
can we call it `sideward_inflation_scaling_factor_`?

on the other PR I mentioned that `scale` is very generic;  we can invent a fancy name like "ahead_clearance_scale",,,  or something less pompous

Reply:
I named it like that since I was thinking we apply this new scale to the `max_sideward_inflation` parameter, and then apply the regular scaling on top. This is how it worked before these PRs. But I guess now you can think of it the other way -- we apply this new scale to the result of the regular scale x `max_sideward_inflation` param.

Anyway, renamed `max_sideward_inflation_scale` to `sideward_inflation_scale`.

I think the names are fine. `scale` is the overall scaling factor that applies to both forward and sideward inflation.
`sideward_inflation_scale` applies specifically only to sideward inflation on top of the regular scaling.

Between `scale` and `scaling factor` I'd rather go with the former -- they mean the same thing but less letters to type. Plus, we keep code changes to a minimum.